src/kalshi.rs

```diff
@@ -0,0 +1,377 @@
+use anyhow::{Context, Result};
+use chrono::{DateTime, Duration, Utc};
+use log::{debug, info};
+use reqwest::blocking::Client;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::settings::{KalshiQuestionRequirements, Settings};
+use crate::types::{BinaryResolution, Question, QuestionSource};
+
+fn list_questions(
+    client: &Client,
+    params: KalshiListQuestionsParams,
+) -> Result<KalshiQuestionsResponse> {
+    debug!("kalshi::list_questions called"); // (params: {:?})", params);
+    Ok(client.get("https://trading-api.kalshi.com/v1/events/")
```

Comment:
Where is this API documented? The docs I found are `/v2/`.

Reply:
Is this endpoint paginated?

Reply:
They're both paginated, but I initially didn't realize that for v1, and can fix that. I haven't found any documentation for v1, but it's what the frontend uses, and has more complete data.

Trouble with v2:
 - I can't get the [markets](https://trading-api.kalshi.com/trade-api/v2/markets?limit=10) endpoint to work for listing markets, it always says I'm missing parameters. I can get an [individual market](https://trading-api.kalshi.com/trade-api/v2/events/JOBLESS-21AUG28), but no resolution criteria
 - the [events](https://trading-api.kalshi.com/trade-api/v2/events?limit=10&with_nested_markets=true) endpoint also has empty resolution criteria, similarly when an individual event specified.